curator-recipes/src/main/java/org/apache/curator/framework/recipes/locks/Reaper.java

```diff
@@ -333,7 +355,20 @@ public void notLeader()
                 reapingIsActive.set(false);
             }
         };
-        localLeaderLatch.addListener(listener);
-        return localLeaderLatch;
+        leaderLatch.addListener(listener);
+
+        reapingIsActive.set(leaderLatch.hasLeadership());
```

Comment:
Another potential race condition here? If the Reaper gets passed a LeaderLatch in its constructor that has already undergone election (and been elected leader) then the isLeader() method on the listener will not fire until another election occurs. While the reapiningIsActive flag will be set to true, the schedule() method will not get called so no reaping will actually occur?